src/LabelMatcher.ts

```diff
@@ -53,6 +82,12 @@ export class LabelMatcher {
                     return this.startsWith(config.labelRule as ILabelStartsWithRule);
                 case "values":
                     return this.contains(config.labelRule as ILabelEqualsRule);
+                case "not":
+                    return this.not(config.labelRule as ILabelNotRule);
+                case "doesntStartWith":
+                    return this.doesntStartWith(config.labelRule as ILabelDoesntStartWithRule);
+                case "doesntEndWith":
+                    return this.doesntEndWith(config.labelRule as ILabelDoesntEndWithRule);
```

Comment:
Hi @Panaetius I'm conflicted with the implementation, 
I think if we just did: `!this.contains`, `!this.endsWith` and `!this.startsWith` then we wouldn't have needed all other rules and tests, we could just add a test for LabelMatcher instead,

Can you do that instead? Just remove the other lines, update this block of code, then add 3 tests to cover the not part,

Reply:
I tried to do that originally, but since the input of e.g. `contains` has to implement `ILabelEqualsRule` which in turn defined the `values` property, we cannot just call it with some other input that's under a different rule. Or maybe I'm missing something